### ISSUE TITLE
DeliveryMethod is immutable

### ISSUE THREAD

**Issue by rainerborene** [readysteady/mailgunner#22]
Hey,

I'm trying to change the `mailgun_settings` on `after_action` callback but after looking at the code it seems that it does not respect the `settings` data and convention of `DeliveryMethod` class.

```ruby
class UserMailer < ApplicationMailer
  before_action { @business, @user = params[:business], params[:user] }
 
  after_action :set_delivery_options
 
  private
    def set_delivery_options
      if @business && @business.has_mailgun_settings?
        mail.delivery_method.settings.merge!(@business.mailgun_settings)
      end
    end
end
```

**Comment by rainerborene** [comment]
I've decided to monkey patch for now.


```ruby
require "mail/check_delivery_params"

class Mailgun::DeliveryMethod
  attr_reader :settings

  def initialize(values)
    @settings = values
  end

  def deliver!(mail)
    check(mail)
    client.send_mime(mail)
  end

  private
    def check(mail)
      Mail::CheckDeliveryParams.check(mail)
    end

    def client
      Mailgunner::Client.new(settings)
    end
end
```

**Comment by timcraft** [comment]
Fixed in d6a00d0 (available in v3.1.0)

**Comment by rainerborene** [comment]
Thanks!
